### DISCUSSION TITLE
Checking habits for past days should not clear current sticky notifications

### DISCUSSION THREAD

**Discussion by Noracsa2000** [iSoron/uhabits#1462] [Feature Requests]
## Pre-submission checklist
- [x] I am submitting a bug report, not a feature request.
- [x] I have searched for similar issues, but did not find any matches.

## Description

There is one imo illogical behaviour, which has been introduced somewhere between the old 1.8 version and the current 2.0.3. It might be related to the solution for #855.

When having a daily habit with a notification at 10 A.M., which has not been checked, any checking of any (even long) past day wrongly clears the current notification.

But the purpose of the notification is to remind me of doing the habit. So any checking for previous days should only clear the notification, if the habit, now, does not need to be done anymore.

So, in this example, the notification for a daily habit at 10 A.M. should only be cleared, if I check yesterday's occurence and it's still earlier then 10 A.M. today.

Checking days before yesterday should not affect yesterday's notification, as yesterday's habit has not been done.

And checking yesterday's occurrence after today 10 A.M. should also not clear the notification, as this notification is now related to today's occurrence, which has not been done yet.

In general, I think, checking a habit for any past day should trigger a verification, if there is a more current "reason" for the notification to be there. (That is, any newer occurrence, which should have been done, but hasn't.) Only if this is not the case, the current notification should be cleared.

## Steps to reproduce
1. Create a daily habit with a notification at 10 A.M. or any other time that suits
2. After the notification has popped up, wait till the next day (waiting only necessary for step 4)
3. Checking the habit for any day before yesterday wrongly clears yesterday's notification
4. (Alternative to 3.) Checking yesterday's habit after the notification time of today wrongly clears today's notification

## System information
 - Phone: Google Pixel 3
 - Phone Operating System: Android 11
 - App version: 2.0.3

**Comment by iSoron** [comment]
Thanks for the report, @Noracsa2000. I'm moving it to feature requests since this is intended app behavior, not something unexpected to the developers. If anyone else thinks this behavior should change, please upvote.
